src/lua/LuaObject.cpp

```diff
@@ -562,23 +560,52 @@ void LuaObjectBase::Register(LuaObjectBase *lo)
 	LUA_DEBUG_END(l, 0);
 }
 
-void LuaObjectBase::Deregister(LuaObjectBase *lo)
+void LuaObjectBase::RegisterPersistent(LuaObjectBase *lo)
 {
-	LuaWrappable *o = lo->GetObject();
-
 	lua_State *l = Lua::manager->GetLuaState();
+	LUA_DEBUG_START(l); // lo userdata
+
+	Register(lo);
```

Comment:
It seems to me that the fact that the object gets into both registers, and then from one of them is deleted by chance - is not very sane. I don't know, maybe do something like `Register(l, "LuaObjectPersistentRegistry")` ?

Reply:
The persistent registry just exists to prevent the LuaObject from being garbage collected when it's no longer referenced directly by Lua - we still have to register in the transient registry to allow the rest of the LuaObject system to work properly. If I could make it work in a single registry table I'd be happy, but the need to support RefCounted / GC'd objects as well as persistent C++ owned objects means it has to be done this way.

The object will not be removed from the transient registry while a reference is held in the persistent registry - the only time something is deleted from the transient registry by chance is during a garbage collection cycle when a LuaSharedObject / LuaCopyObject is cleaned up due to a lack of references.

Reply:
oops, I really misunderstood how it works, sorry!